metaflow/client/live_run.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+
+class LiveRun:
+    """
+    This object allows users to access information relating to the run it
+    represents, including booleans for whether the run has been triggered, is
+    running, and was successful, as well as for failed steps, and exceptions.
+
+    However, this class is not meant to be instantiated; instead, each plugin
+    or under layer should have its own class to represent a live run and each
+    such class should inherit from this class.
+    """
+
+    def __init__(self):
+        """
+        Initialize a LiveRun object. This class is a parent class only and is
+        not intended to be instantiated.
+
+        """
+        self._flow_name = None
+        self._metaflow_run_id = None
+
+    @classmethod
+    def trigger(cls, **kwargs) -> LiveRun:
+        raise NotImplementedError(
+            "Error occurred because either LiveRun class was"
+            "instantiated or child class did not override method"
+        )
+
+    @property
+    def flow_name(self) -> str:
+        return self._flow_name  # TODO: raise exception if None???
+
+    @property
+    def run_id(self) -> str:
+        return self._metaflow_run_id  # TODO: raise exception if None???
```

Comment:
Is there a case where you have a live run class but do not have `flow_name` or `run_id`?

Reply:
It depends. 

If the user uses the classmethod trigger to create a LiveRun object (or a child of LiveRun), then there will always be both a `flow_name` and `run_id` because trigger populates them with the correct info before returning (regardless of whether wait is turned on).

However, if the user creates a LiveRun object (or a child of LiveRun) without using the trigger method, then they will not always be populated.

My sense is that we do not need to add an exception here because this use case will never be able to trigger and will generally fail.

Reply:
Make sense. Since you already explained your decision here, would it make sense to remove the `TODO` comment then? Feel free to add doc string if you feel like anything should be documented as well.

Reply:
I will remove TODO and add a brief docstring to explain